src/adapter/src/coord/ddl.rs

```diff
@@ -508,7 +508,7 @@ impl Coordinator {
                 }
             }
             if !sources_to_drop.is_empty() {
-                self.drop_sources(sources_to_drop);
+                self.drop_sources(sources_to_drop).await;
```

Comment:
This is just "infected" with async because executing some sources requires async (though this does not). Will amend with a comment that this needs to have the same behavior as `create_collections`, which doesn't run in a task.